### DISCUSSION TITLE
integrating/switching-channels

### DISCUSSION THREAD

**Discussion by giscus** [velopack/velopack.docs#20] [General]
# integrating/switching-channels

By default, UpdateManager will try and search for releases in the same channel that it was built for. You can read more about packaging channels here. So normally, you should not provide a channel at all to the UpdateManager constructor.

https://docs.velopack.io/integrating/switching-channels

**Comment by framlingham** [comment]
This sounds great - I've wanted to do exactly this, so will be trying out Velopack. Sounds like you've made a great framework! I'll try to remember to check back in once we've migrated.

**Comment by belidzs** [comment]
Is it possible to somehow promote a release from beta to stable?

I guess at the very minimum I have to run `vpk pack` twice on the same `packDir` content to create two separate releases.

If that's the case do I also have to make sure that `packVersion` is different for these two releases or can they remain the same? (an example would be `1.0.0` and `1.0.0-beta.0`)

**Reply by caesay** [reply]
At the moment you can not "promote" a release using the velopack tooling, but you can create the same version on multiple channels. Channels operate completely independently and get their own unique release feed (releases.json file).

**Reply by belidzs** [reply]
Thanks, I eventually made this work by realizing that in this case (when you're switching to the same version on a different channel) you *must* apply the changes manually, it won't pick them up automatically on application restart.

```
mgr.ApplyUpdatesAndRestart(newVersion);
```
